curriculum/challenges/english/20-upcoming-python/learn-tree-traversal-by-building-a-binary-search-tree/65560f9380be92226084ef46.md

```diff
@@ -0,0 +1,44 @@
+---
+id: 65560f9380be92226084ef46
+title: Step 1
+challengeType: 20
+dashedName: step-1
+---
+
+# --description--
+
+In this project, we would create a Binary Search Tree. A binary search tree is a data structure in which each node has at most two children, with the left child containing values less than the node and the right child containing values greater than the node, allowing for efficient searching and sorting operations.
+
+Begin by defining an empty class `TreeNode`. The `TreeNode` class represents a `node` in a binary search tree(BST).
+
+# --hints--
+
+You should use the `class` keyword to define a class.
+
+```js
+({ test: () => assert.match(code, /class/) })
+```
+
+Your class name should be `TreeNode`.
+
+```js
+({ test: () => assert.match(code, /class\s+TreeNode/) })
+```
+
+Don't forget to add a `:` at the end.
+
+```js
+({ test: () => assert.match(code, /class\s+TreeNode\s*:/) })
+```
+
```

Comment:
@Dario-DC these tests(normal version) are also working.

You should use the `class` keyword to define a class.

```js
assert.match(code, /class/);
```

Your class name should be `TreeNode`.

```js
assert.match(code, /class\s+TreeNode/);
```

Don't forget to add a `:` at the end.

```js
assert.match(code, /class\s+TreeNode\s*:/);
```

Reply:
Maybe we can reduce the number of tests for this project, considering that this should be towards the end of the curriculum. Something like:

You should use the `class` keyword to declare an empty class named `TreeNode`. Don't forget to add the colon at the end.

```js
({
  test: () => {
    assert(__pyodide.runPython(`
      import inspect
      a = __locals.get("TreeNode")
      inspect.isclass(a)
    `));
  }
})
```